transformer_lens/utils.py

```diff
@@ -807,3 +807,141 @@ def override_or_use_default_flag(
     Otherwise, the global flag is returned.
     """
     return override if override is not None else default_flag
+
+
+def get_attention_mask(
+    tokenizer,
+    tokens: Int[torch.Tensor, "batch pos"],
+    prepend_bos: bool,
+) -> Int[torch.Tensor, "batch pos"]:
+    """
+    This method calculates an attention mask for a given sequence of tokens when left padding is used. The attention
+    mask is a binary tensor where each element is 1 if the corresponding token is not a pad token and 0 otherwise. In
+    the special case where the BOS token is the same as the pad token and bos token is prepended to process the tokens,
+    this method ensures that the prepended BOS (pad) token also gets a 1 in the attention mask.
+    Gotcha: Make sure that the passed prepend_bos is the same as the one used to generate the tokens!
+    """
+    # Initialize the attention mask to be 1 wherever the token is not a pad token
+    attetnion_mask = tokens.ne(tokenizer.pad_token_id)
+
+    # Handle the special case where the BOS token is the same as the pad token
+    if tokenizer.bos_token_id == tokenizer.pad_token_id and prepend_bos:
```

Comment:
@neelnanda-io Hi Neel, the special case that you just pointed out is handled here!